userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -21,6 +29,51 @@ public class UserServiceImpl implements IUserService {
     public UserServiceImpl(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
+    @Async  // Annotation cho bất đồng bộ
+    public CompletableFuture<Void> importUsersFromCsv(MultipartFile file) throws IOException {
+        List<User> usersToSave = new ArrayList<>();
+
+        // Đọc file CSV
+        try (CSVReader reader = new CSVReader(new InputStreamReader(file.getInputStream()))) {
```

Comment:
Using with BufferedReader for better memory management 
```suggestion
        try (CSVReader reader = new CSVReader(new BufferedReader(new InputStreamReader(new FileInputStream("emps.csv"), "UTF-8")), ',');
)) {
```